src/RefractAPI.cc

```diff
@@ -332,63 +380,77 @@ namespace drafter {
         return element;
     }
 
-    refract::IElement* ActionToRefract(const snowcrash::Action& action)
+    refract::IElement* ActionToRefract(const SectionInfo<snowcrash::Action>& action)
     {
         refract::ArrayElement* element = new refract::ArrayElement;
         RefractElements content;
 
         element->element(SerializeKey::Transition);
-        element->meta[SerializeKey::Title] = refract::IElement::Create(action.name);
+        element->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_SECTION_INFO(action, name));
 
-        if (!action.relation.str.empty()) {
-            element->attributes[SerializeKey::Relation] = refract::IElement::Create(action.relation.str);
+        if (!action.section.relation.str.empty()) {
+            // FIXME: add SourceMap
+            element->attributes[SerializeKey::Relation] = refract::IElement::Create(action.section.relation.str);
         }
 
-        if (!action.uriTemplate.empty()) {
-            element->attributes[SerializeKey::Href] = refract::IElement::Create(action.uriTemplate);
+        if (!action.section.uriTemplate.empty()) {
+            // FIXME: This doues not chnge rendering, do we have exammple with uriTemplate?
+            element->attributes[SerializeKey::Href] = PrimitiveToRefract(MAKE_SECTION_INFO(action, uriTemplate));
         }
 
-        if (!action.parameters.empty()) {
-            element->attributes[SerializeKey::HrefVariables] = ParametersToRefract(action.parameters);
+        if (!action.section.parameters.empty()) {
+            element->attributes[SerializeKey::HrefVariables] = ParametersToRefract(MAKE_SECTION_INFO(action, parameters));
         }
 
-        if (!action.attributes.empty()) {
-            refract::IElement* dataStructure = DataStructureToRefract(action.attributes);
+        if (!action.section.attributes.empty()) {
+            refract::IElement* dataStructure = DataStructureToRefract(MAKE_SECTION_INFO(action, attributes));
             dataStructure->renderType(refract::IElement::rFull);
             element->attributes[SerializeKey::Data] = dataStructure;
         }
 
-        content.push_back(CopyToRefract(action.description));
+        content.push_back(CopyToRefract(MAKE_SECTION_INFO(action, description)));
+
+        typedef SectionInfoCollection<snowcrash::TransactionExamples> ExamplesType;
```

Comment:
I do not see problem to call it Type. It is better to know it is type later in code in first look.